cmd/minishift/cmd/config/config.go

```diff
@@ -100,6 +100,14 @@ var settings []Setting = []Setting{
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
+		name: "hostfolders-automount",
```

Comment:
> For the homefolder I can understand this to be on global level

:-)

> $ minishift hostfolder add Projects --global --username="dsd" --password="fdfs" ...
> $ minishift hostfolder add Disks --instance=minishift --username="dsd" --password="fdfs" ...

So you are suggestion to have the possibility to add host folders on a global level (aka, the config persist even after _minishift delete_) and one instance level. I guess that works. It gets a bit more complicated, but if you feel strongly about having host folders on an instance level as well, I guess this works.

I am not sure whether _global_ is the right option name. Maybe something like '--all-vms' or '--instance-only'. Not sure. this could also be a single boolean flag. in this case. Regarding '--instance=minishift', I don't think one needs to specify the instance here. If anything it would just be '--instance'. The current thinking on multiple instances is more around that one has a similar behaviour as with _oc_. One sets a current instance 'minishift use foo' and from then on all commands are specific to the 'foo' instance. On top of this, one could have a global  '--instance-name' options to execute a single command against the specified instance. In both cases, naming the instance as part of the hostfolder sub-command would not be required. 

+1 for having an option for determining the "scope" of the hostfolder. I think int this case we get away with a single boolean flag.